### ISSUE TITLE
Distributing computations over four high end GPU's

### ISSUE THREAD

**Issue by ChrisHonsinger** [helix-toolkit/helix-toolkit#1343]
We have 4 AMD WX 9100s (GPUs).  We have multiple display rectangles configured for each of the four GPUs using AMD's Eyefinity.  

We started the project using Helix.WPF library. WPF will select the GPU associated with the window rectangles created for the viewport automatically. 

Helix.WPF wasn't fast enough, so now we are using Helix.WPF.SharpDX. Problem now is that 98% of one of the four GPUs is used while the remaining three GPUs use ~3-9%.

How can I fix this?

Chris

**Comment by holance** [comment]
You may want to assign different adapter index when creating `EffectsManager`.

**Comment by ChrisHonsinger** [comment]
Holance - I initialized the Effectsmanager class with the adapter index (as is calculated within GetBestAdapter) and still the same GPU is at 95% and the other three are at ~5%.

**Comment by holance** [comment]
GetBestAdapter will return same index if you have multiple gpu with same config, you need to specify different index manually for different effects manager.

**Comment by ChrisHonsinger** [comment]
I recognized that. I used the GetBestAdapter example code to ensure the index was meaningful to use is the EffectsManager initialization.

**Comment by ChrisHonsinger** [comment]
I tried using EnableSwapChainRendering="True". GPU usage on the heavily taxed GPU( mentioned above) went down by 50%. However, another problem ensued.

Each of the four GPUs (above) drives six displays. The GPU displays run WPF.SharpDX in different processes (not threads). As the processes get toward the last few displays, the monitors go in and go out several times and then the system freezes. 

I am timing the camera update for each display using the WPF CompositionRendering event. Given that the Viewport3D uses a Windows Forms type window, I am wondering if that may be getting in the way. 

Also, I tried to lower the frame rate by using  FrameRate="30" - that didn't work.

Any ideas what I need to look at?

Thanks very much.
